src/ng/directive/select.js

```diff
@@ -183,15 +185,12 @@ var SelectController =
     } else if (interpolateValueFn) {
       // The value attribute is interpolated
       optionAttrs.$observe('value', function valueAttributeObserveAction(newVal) {
-        var currentVal = self.readValue();
```

Comment:
Yes, but I think we are overwriting the default implementation in `ngOptions` ([ngOptions.js#L506][1]).
I am not sure that there is indeed a usecase, where the side-effects in the new `readValue()` are necessary.

[1]: https://github.com/mgol/angular.js/blob/9bb014b44db3c46bf9a018d4d859fa63bb60a585/src/ng/directive/ngOptions.js#L506